include/lbann/data_readers/sample_list_open_files_impl.hpp

```diff
@@ -548,13 +549,12 @@ ::all_gather_packed_lists(lbann_comm& comm) {
         if(m_file_map.count(filename) == 0) {
           m_file_map[filename] = file_map.at(filename);
         }
+        mp[filename] = index;
       }else {
-        for(size_t i = 0; i < m_file_id_stats_map.size(); i++) {
-          if(filename == get_samples_filename(i)) {
-            index = i;
-            break;
-          }
+        if (mp.find(filename) == mp.end()) {
+          LBANN_ERROR("mp.find(filename) == mp.end()");
         }
+        index = mp[filename];
```

Comment:
This essentially does the lookup twice. GH finally has multiline comments, but not multiline "suggestions", so here's a codeblock instead:
```cpp
auto search_result = mp.find(filename);
if (search_result == mp.end()) {
  LBANN_ERROR("mp.find(filename) == mp.end()");
}
index = search_result->second;
```

Reply:
As soon as @benson31 fix is applied, it looks good.